.eslintrc.json

```diff
@@ -1,8 +1,112 @@
 {
-  "extends": ["next/core-web-vitals", "prettier"],
-  "plugins": ["prettier"],
-  "ignorePatterns": ["**/dist/*.js", "/docs/**", "/out/**"],
+  "plugins": ["prettier", "@typescript-eslint", "import"],
+  "parserOptions": {
+    "project": "./tsconfig.json"
+  },
+  "parser": "@typescript-eslint/parser",
+  "extends": [
+    "plugin:@typescript-eslint/strict-type-checked",
+    "plugin:@typescript-eslint/stylistic-type-checked",
+    "plugin:import/recommended",
+    "plugin:import/typescript",
```

Comment:
* `@typescript-eslint` focuses on TS-specific checks
* `import` focuses on modules

Both include a ton of ultra useful defaults, so we just opt-in or configure some options below